packages/core/src/graph/create-dependencies.ts

```diff
@@ -40,49 +47,123 @@ export const createDependencies: CreateDependenciesCompat<
+        const newDeps: RawProjectGraphDependency[] = [];
+        const { ext } = parse(file.file);
+        if (['.csproj', '.fsproj', '.vbproj'].includes(ext)) {
+          if (!dotnetClient) {
+            console.warn('Dotnet client not initialized');
+            return newDeps;
+          }
+          try {
+            const references = await dotnetClient.getProjectReferencesAsync(
+              file.file,
+            );
+            for (const reference of references) {
+              // Skip empty, null, or undefined references
+              if (
+                !reference ||
+                typeof reference !== 'string' ||
+                reference.trim() === ''
+              ) {
+                continue;
+              }
+
+              const project = resolveReferenceToProject(
+                reference,
+                file.file,
+                rootMap,
+                ctx,
+              );
+              if (project) {
+                newDeps.push({
+                  source,
+                  target: project,
+                  type: DependencyType.static,
+                  sourceFile: file.file,
+                });
+              } else {
+                console.warn(
+                  `Unable to resolve project for reference ${reference} in ${file.file}`,
+                );
+              }
+            }
+          } catch (error) {
             console.warn(
-              `Unable to resolve project for reference ${reference} in ${file.file}`,
+              `Failed to get project references for ${file.file}:`,
+              error,
             );
           }
         }
-      }
-      return newDeps;
+        return newDeps;
+      };
+      const getAllProjectReferences = changed.map(getProjectReferences);
+      return Promise.all(getAllProjectReferences).then((d) => d.flat());
     };
-    const getAllProjectReferences = changed.map(getProjectReferences);
-    return Promise.all(getAllProjectReferences).then((d) => d.flat());
-  };
 
-  const parseAllProjects = Object.keys(ctx.filesToProcess.projectFileMap).map(
-    parseProject,
+    const parseAllProjects = Object.keys(ctx.filesToProcess.projectFileMap).map(
+      parseProject,
+    );
+    return Promise.all(parseAllProjects).then((d) => d.flat());
+  })();
+
+  // Add a timeout to the entire operation to prevent infinite loops
+  const timeoutPromise = new Promise<RawProjectGraphDependency[]>(
+    (_, reject) => {
+      const timeoutId = setTimeout(() => {
+        reject(new Error('Dependency resolution timed out after 30 seconds'));
+      }, 30000);
+
+      // Use unref() to prevent the timeout from keeping the process alive
+      timeoutId.unref();
+    },
   );
-  return Promise.all(parseAllProjects).then((d) => d.flat());
+
+  try {
+    return await Promise.race([dependencyResolutionPromise, timeoutPromise]);
+  } catch (error) {
+    console.warn('Failed to resolve .NET project dependencies:', error);
+    return [];
+  }
```

Comment:
We can't do this. If .NET dependency resolution errors, and in CI people are running dotnet test via `nx affected -t test` or similar, they would miss tests that need to run. Affected wouldn't identify projects correctly, dependent builds could be missing etc.

The project graph has to be completely calculated every time CI runs, if there's something that makes that not work x% of the time, we can't patch that by ignoring the .NET dependencies x% of the time, because that leads to invalid CI states.